apps/web/src/pages/SupportPage/components/SupportForm/SupportForm.tsx

```diff
@@ -274,15 +274,36 @@ const SupportFormApi = () => {
       );
       return Promise.reject(response.status);
     });
+
+  const dataToUser = (input: GetMeQuery): User | null => {
+    if (input) {
+      if (input.me) {
+        return {
+          ...input.me,
+          poolCandidates: input.me.poolCandidates?.map((candidate) => {
+            if (candidate) {
+              return {
+                ...candidate,
+                pool: { id: candidate.poolAdvertisement?.id || "" },
+              };
+            }
+            return null;
+          }),
+        };
+      }
+    }
+    return null;
+  };
   const [{ data, fetching, error }] = useGetMeQuery();
+  const userData = data ? dataToUser(data) : null;
   const [showSupportForm, setShowSupportForm] = React.useState(true);
 
   return (
     <Pending fetching={fetching} error={error}>
       <SupportForm
         showSupportForm={showSupportForm}
         onFormToggle={setShowSupportForm}
-        currentUser={data?.me ?? null}
+        currentUser={userData}
```

Comment:
The SupportForm does not actually need a whole user object. It only uses a few fields. Instead of calling dataToUser, I think you can replace SupportFormProps.currentUser with `currentUser?: Pick<User, "firstName" | "lastName" | "email"> | null;`

In fact, this page should be using a much smaller query than getMe, but that's out of scope.

Reply:
Huh, I didn't really look at what fields this was actually using. Changed with [529108b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5927/commits/529108bb1075c579ea4b05b120f56796402d4492).